### PR TITLE
Add initial version of new release script

### PR DESCRIPTION
**Pull request by otoolep**
WIP, nowhere near ready for merging.

### CODE REVIEW

**Comment by sanga** [comment]
Are we tied to fpm? As a debian user, it feels clunky to have to install an external tool to create a deb. Much nicer in my opinion would be to have the /debian directory with all the package metadata and build instructions (and /rpm or whatever is appropriate for Redhat). I realise that this means you need to build a deb package on debian and a rpm on redhat/fedora but that shouldn't be too inconvenient should it?

**Comment by otoolep** [comment]
@sanga -- no, we're not wedded to fpm. While it is an external dependency, it greatly simplifies the workflow once it is available. I would like to hear more about your ideas for other directories, and precisely what you mean by "metadata". If you wanted to generate a PR to show me what you mean, I'd certainly take a look.

**Comment by otoolep** [comment]
OK, script updated to check that the proposed version is not already a tag, and packaging actually enabled. When I run this on my box, which has `fpm` installed I get an RPM and a Debian package.

**Comment by otoolep** [comment]
Now ready for merging.

File: package.sh [hunk 1]
```diff
@@ -0,0 +1,242 @@
+#!/bin/bash
+
+###########################################################################
+# Packaging script which creates debian and RPM packages. It optionally
+# tags the repo with the given version. 'fpm' must be on the path.
+
+INSTALL_ROOT_DIR=/opt/influxdb
+CONFIG_ROOT_DIR=/etc/opt/influxdb
```
Comment (toddboom) [inline package.sh]: is this more "standard" than `/etc/influxdb`?

File: scripts/init.sh [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            if [ "x$status" = "x0" ]; then
+                log_failure_msg "$name process is running"
+                exit 1 # Exit
+            fi
+        fi
+        # Start the daemon.
+        log_success_msg "Starting the process" "$name"
+        # Start the daemon with the help of start-stop-daemon
+        # Log the message appropriately
+        cd /
```
Comment (toddboom) [inline scripts/init.sh]: it looks like you're using absolute paths everywhere - is this just for consistency?

**Comment by toddboom** [comment]
This looks awesome. Nice work!

File: package.sh (hunk 1 above)
Comment (otoolep) [inline package.sh]: Technically yes, according to the Filesystem Hierarchy standard: http://refspecs.linuxfoundation.org/FHS_2.3/fhs-2.3.pdf

See section 3.7.4

Comment (otoolep) [inline package.sh]: To be clear, we do this because our software is being installed in `/opt`.

File: scripts/init.sh (hunk 2 above)
Comment (otoolep) [inline scripts/init.sh]: Actually, this file was copied (with some small clean-up) from our existing `init.d` script. I have yet to dig into it very closely, but wanted to use it as a starting point.

**Comment by otoolep** [comment]
Merging now.

**Comment by sanga** [comment]
@otoolep if you don't have fpm, and you're on debian then it's a hassle to install fpm (and ruby and rubygems in order to get fpm). What I thought was something like how it's done here: https://github.com/python-diamond/Diamond (see /debian and /rpm there). Then if you're on a debian system you just use the standard tools (and similar on a fedora system). That of course means you need a debian and a fedora system to build though

**Comment by otoolep** [comment]
@sanga -- thanks for the reference. I'll take a look should we decide to modify our packaging process further. As you say, you do need specific systems to build specific packages, which is exactly what we like to avoid -- and do with fpm.

**Comment by sanga** [comment]
well, I had another go at packaging on deb wheezy, with significantly more success than last time. Not quite sure what happened last time but I wound up with a huge mess. Anyway this time round I got a proper package without any hassle. So I guess we could live fpm then. I've a few comments about the package itself but I'll start that in a bug I think.

**Comment by otoolep** [comment]
Please do @sanga -- feedback welcome.

**Pull request merged**